### PR TITLE
LPD-36817 - Account api endpoints do not handle empty organizationId/ERC arrays correctly

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-36817

Test already covered by LocalFile.CommerceOrganizationManagementWidget#CanUserRemoveAccountFromOrganization

### CODE REVIEW

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by brbalazs** [comment]
ci:reevaluate:1377401_17947

**Comment by brbalazs** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 1]
```diff
@@ -1055,36 +1055,43 @@ private Page<Account> _getOrganizationAccountsPage(
 
 	private long[] _getOrganizationIds(Account account) {
 		Long[] organizationIds = account.getOrganizationIds();
+		String[] organizationExternalReferenceCodes =
+			account.getOrganizationExternalReferenceCodes();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java:1059]: swipe for order

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -1055,36 +1055,43 @@ private Page<Account> _getOrganizationAccountsPage(
...
 
-		if (!ArrayUtil.isEmpty(organizationIds)) {
+		if ((organizationIds == null) &&
+			(organizationExternalReferenceCodes == null)) {
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java:1062]: swipe for order

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 3]
```diff
@@ -1055,36 +1055,43 @@ private Page<Account> _getOrganizationAccountsPage(
...
+
+			return null;
+		}
+
+		if ((organizationIds != null) && (organizationIds.length > 0)) {
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java:1067]: !ArrayUtil.isEmpty(organizationIds) should be enough

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java [hunk 4]
```diff
@@ -1055,36 +1055,43 @@ private Page<Account> _getOrganizationAccountsPage(
...
 			return ArrayUtil.toArray(organizationIds);
 		}
 
-		String[] organizationExternalReferenceCodes =
-			account.getOrganizationExternalReferenceCodes();
+		if ((organizationExternalReferenceCodes != null) &&
+			(organizationExternalReferenceCodes.length > 0)) {
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/AccountResourceImpl.java:1072]: !ArrayUtil.isEmpty(organizationExternalReferenceCodes) should be enough

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:um

**Comment by smottal** [comment]
ci:report:16302098

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**